batch.go

```diff
@@ -0,0 +1,177 @@
+package nimbusdb
+
+import (
+	"bytes"
+	"sync"
+	"time"
+
+	"github.com/manosriram/nimbusdb/utils"
+)
+
+type Batch struct {
+	db         *Db
```

Comment:
add a closed flag with `false` as default value setting it to `true` on closing.